terragrunt/aws/cloudfront/iam.tf

```diff
@@ -0,0 +1,52 @@
+resource "aws_iam_role" "waf_log_role" {
+  name               = "simplify-privacy-app-logs"
+  assume_role_policy = data.aws_iam_policy_document.firehose_assume_role.json
+
+  tags = {
+    CostCentre = var.billing_code
+    Terraform  = true
+  }
+}
+
+data "aws_iam_policy_document" "firehose_assume_role" {
+  statement {
+    effect = "Allow"
+
+    principals {
+      type        = "Service"
+      identifiers = ["firehose.amazonaws.com"]
+    }
+
+    actions = ["sts:AssumeRole"]
+  }
+}
+
+resource "aws_iam_policy" "write_waf_logs" {
```

Comment:
You'll also need a policy attachment to attach the `write_waf_logs` policy to the `waf_log_role` role:
https://github.com/cds-snc/scan-files/blob/5b153e30dbeb62fef5540125b6b1f95409ebc9ee/terragrunt/aws/api/iam.tf#L183-L186